File: CVRP/source/utils/LocationNode.cpp
#include "LocationNode.h"

LocationNode::LocationNode() = default;

LocationNode::LocationNode(int ID, int demand)
{
    locationID = ID;
    locationDemand = demand;
    xCoord = 0;
    yCoord = 0;
    visited = 0;
}

LocationNode::LocationNode(int ID, int x, int y)
{
    locationID = ID;
    xCoord = x;
    yCoord = y;
    locationDemand = 0;
    visited = 0;
}

LocationNode::LocationNode(int ID, int x, int y, int demand)
{
    locationID = ID;
    xCoord = x;
    yCoord = y;
    locationDemand = demand;
    visited = 0;
}

int LocationNode::GetID() const
{
    return locationID;
}

int LocationNode::GetDemand() const
{
    return locationDemand;
}

int LocationNode::GetX() const
{
    return xCoord;
}

int LocationNode::GetY() const
{
    return yCoord;
}

int LocationNode::TimesVisited()
{
    return visited;
}

void LocationNode::SetDemand(int demand)
{
    locationDemand = demand;
}

void LocationNode::SetCoordinates(int x, int y)
{
    xCoord = x;
    yCoord = y;
}

void LocationNode::Visit()
{
    visited += 1;
    //std::cout << "Visited Node " << locationID << ": visit no. " << visited << std::endl;
}

File: py_CVRP/classes/LocationNode.py
class LocationNode:

    def __init__(self, node_id, x_coord=0, y_coord=0, demand=0):
        self.loc_id = node_id  # ID of the location (= vrp_ID - 1 = sol_ID)
        self.demand = demand
        self.x = x_coord
        self.y = y_coord
        self.times_visited = 0
        self.route_ids = []     # List of routes that have visited this node
                                # (should be one for every node except depot)
        self.average_distance = 0

    def visit(self, route_id=0):
        self.times_visited += 1
        self.route_ids.append(route_id)

    def coordinates(self):
        return [self.x, self.y]

    def print(self):
        print("ID:", self.loc_id, " Demand:", self.demand, " Times visited:", self.times_visited)
